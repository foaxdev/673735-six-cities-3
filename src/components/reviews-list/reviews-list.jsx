import React from "react";
import PropTypes from "prop-types";
import {Review} from "../review/review";

export const ReviewsList = ({reviews}) => {
  reviews.sort((a, b) => b.date - a.date);

  const reviewsList = reviews.slice(0, 10).map((review, index) =>
    <Review
      key={review.text + index}
      id={review.id}
      rating={review.rating}
      avatar={review.avatar}
      date={review.date}
      name={review.name}
      text={review.text}
    />
  );

  return (
    <ul className="reviews__list">
      {reviewsList}
    </ul>
  );
};

ReviewsList.propTypes = {
  reviews: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      avatar: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      rating: PropTypes.number.isRequired,
      date: PropTypes.number.isRequired,
      text: PropTypes.string.isRequired
    }).isRequired
  ).isRequired
};
