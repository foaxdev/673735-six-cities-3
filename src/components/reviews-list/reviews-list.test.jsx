import React from 'react';
import renderer from 'react-test-renderer';
import {ReviewsList} from "./reviews-list";

const TEST_REVIEWS = [
  {
    id: 0,
    avatar: `img/avatar-max.jpg`,
    name: `Max`,
    rating: 5,
    date: 1082368261115,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  },
  {
    id: 1,
    avatar: `img/avatar-max.jpg`,
    name: `Anna`,
    rating: 4,
    date: 1500368261000,
    text: `Good place, good food, good everything`
  },
  {
    id: 2,
    avatar: `img/avatar-max.jpg`,
    name: `Akira`,
    rating: 5,
    date: 1580068260000,
    text: `Excellent`
  },
  {
    id: 3,
    avatar: `img/avatar-max.jpg`,
    name: `Miki`,
    rating: 1,
    date: 1082368200000,
    text: `So so, could be better`
  },
  {
    id: 4,
    avatar: `img/avatar-max.jpg`,
    name: `Max`,
    rating: 5,
    date: 1082368261115,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  },
  {
    id: 5,
    avatar: `img/avatar-max.jpg`,
    name: `Anna`,
    rating: 4,
    date: 1500368261000,
    text: `Good place, good food, good everything`
  },
  {
    id: 6,
    avatar: `img/avatar-max.jpg`,
    name: `Akira`,
    rating: 5,
    date: 1580068260000,
    text: `Excellent`
  },
  {
    id: 7,
    avatar: `img/avatar-max.jpg`,
    name: `Miki`,
    rating: 1,
    date: 1082368200000,
    text: `So so, could be better`
  },
  {
    id: 8,
    avatar: `img/avatar-max.jpg`,
    name: `Max`,
    rating: 5,
    date: 1082368261115,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  },
  {
    id: 9,
    avatar: `img/avatar-max.jpg`,
    name: `Anna`,
    rating: 4,
    date: 1500368261000,
    text: `Good place, good food, good everything`
  },
  {
    id: 10,
    avatar: `img/avatar-max.jpg`,
    name: `Akira`,
    rating: 5,
    date: 1580068260000,
    text: `Excellent`
  },
  {
    id: 11,
    avatar: `img/avatar-max.jpg`,
    name: `Miki`,
    rating: 1,
    date: 1082368200000,
    text: `So so, could be better`
  }
];

it(`Should render Reviews List correctly`, () => {
  const tree = renderer
    .create(<ReviewsList reviews={TEST_REVIEWS}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
