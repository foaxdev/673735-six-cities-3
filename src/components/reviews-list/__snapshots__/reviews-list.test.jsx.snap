// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Reviews List correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Akira
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Excellent
      </p>
      <time
        className="reviews__time"
        dateTime="2020-01-26"
      >
        January 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Akira
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Excellent
      </p>
      <time
        className="reviews__time"
        dateTime="2020-01-26"
      >
        January 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Akira
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Excellent
      </p>
      <time
        className="reviews__time"
        dateTime="2020-01-26"
      >
        January 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Anna
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Good place, good food, good everything
      </p>
      <time
        className="reviews__time"
        dateTime="2017-07-18"
      >
        July 2017
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Anna
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Good place, good food, good everything
      </p>
      <time
        className="reviews__time"
        dateTime="2017-07-18"
      >
        July 2017
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Anna
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Good place, good food, good everything
      </p>
      <time
        className="reviews__time"
        dateTime="2017-07-18"
      >
        July 2017
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      </p>
      <time
        className="reviews__time"
        dateTime="2004-04-19"
      >
        April 2004
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      </p>
      <time
        className="reviews__time"
        dateTime="2004-04-19"
      >
        April 2004
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      </p>
      <time
        className="reviews__time"
        dateTime="2004-04-19"
      >
        April 2004
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Miki
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        So so, could be better
      </p>
      <time
        className="reviews__time"
        dateTime="2004-04-19"
      >
        April 2004
      </time>
    </div>
  </li>
</ul>
`;
